File: z-clients/javascript/src/models/idempotencyCreateNamespaceOut.ts
// this file is @generated

export interface IdempotencyCreateNamespaceOut {
    name: string;
    created: Date;
    updated: Date;
}

export const IdempotencyCreateNamespaceOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): IdempotencyCreateNamespaceOut {
        return {
            name: object['name'],
            created: new Date(Number(object['created'])),
            updated: new Date(Number(object['updated'])),
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: IdempotencyCreateNamespaceOut): any {
        return {
            'name': self.name,
            'created': self.created.getTime(),
            'updated': self.updated.getTime(),
        };
    }
}

File: z-clients/rust/src/models/idempotency_create_namespace_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct IdempotencyCreateNamespaceOut {
    pub name: String,

    pub created: u64,

    pub updated: u64,
}

impl IdempotencyCreateNamespaceOut {
    pub fn new(name: String, created: u64, updated: u64) -> Self {
        Self {
            name,
            created,
            updated,
        }
    }
}
